data-raw/vs_peds.R

```diff
@@ -0,0 +1,261 @@
+      VISIT == "WEEK 4" ~ 6.45
+    ),
+    USUBJID == "01-701-1033" & VSTESTCD == "HDCIRC" ~ case_when(
+      VISIT == "SCREENING 1" ~ 40.34,
+      VISIT == "BASELINE" ~ 40.71,
+      VISIT == "WEEK 2" ~ 41.12,
+      VISIT == "WEEK 4" ~ 41.56
+    ),
+    USUBJID == "01-701-1034" & VSTESTCD == "HEIGHT" ~ case_when(
+      VISIT == "SCREENING 1" ~ 53.65,
+      VISIT == "BASELINE" ~ 54.78,
+      VISIT == "WEEK 2" ~ 55.95,
+      VISIT == "WEEK 4" ~ 57.14,
+      VISIT == "WEEK 6" ~ 58.23,
+      VISIT == "WEEK 8" ~ 59.45,
+      VISIT == "WEEK 12" ~ 60.67,
+      VISIT == "WEEK 16" ~ 61.79,
+      VISIT == "WEEK 20" ~ 62.91,
+      VISIT == "WEEK 24" ~ 64.03,
+      VISIT == "WEEK 26" ~ 65.15
+    ),
+    USUBJID == "01-701-1034" & VSTESTCD == "WEIGHT" ~ case_when(
+      VISIT == "SCREENING 1" ~ 4.25,
+      VISIT == "BASELINE" ~ 4.56,
+      VISIT == "WEEK 2" ~ 4.84,
+      VISIT == "WEEK 4" ~ 5.15,
+      VISIT == "WEEK 6" ~ 5.47,
+      VISIT == "WEEK 8" ~ 5.79,
+      VISIT == "WEEK 12" ~ 6.11,
+      VISIT == "WEEK 16" ~ 6.43,
+      VISIT == "WEEK 20" ~ 6.75,
+      VISIT == "WEEK 24" ~ 7.07,
+      VISIT == "WEEK 26" ~ 7.39
+    ),
+    USUBJID == "01-701-1034" & VSTESTCD == "HDCIRC" ~ case_when(
+      VISIT == "SCREENING 1" ~ 38.45,
+      VISIT == "BASELINE" ~ 38.92,
+      VISIT == "WEEK 2" ~ 39.34,
+      VISIT == "WEEK 4" ~ 39.78,
+      VISIT == "WEEK 6" ~ 40.23,
+      VISIT == "WEEK 8" ~ 40.65,
+      VISIT == "WEEK 12" ~ 41.07,
+      VISIT == "WEEK 16" ~ 41.48,
+      VISIT == "WEEK 20" ~ 41.89,
+      VISIT == "WEEK 24" ~ 42.30,
+      VISIT == "WEEK 26" ~ 42.72
+    ),
+    TRUE ~ VSSTRESN
+  ))
+
+# Derive BMI values ----
+vs_peds <- vs_peds %>%
+  arrange(USUBJID, VISITNUM, VSTESTCD, VSDY) %>%
+  group_by(USUBJID, VISITNUM) %>%
+  mutate(
+    VSSTRESN = case_when(
+      VSTESTCD == "BMI" ~ {
+        weight <- VSSTRESN[VSTESTCD == "WEIGHT"]
+        height <- VSSTRESN[VSTESTCD == "HEIGHT"] / 100 # Convert height from cm to m
+        as.numeric(weight / (height^2)) # BMI calculation
+      },
+      TRUE ~ VSSTRESN
+    )
+  ) %>%
+  ungroup()
+
+# Formatting the output dataset ----
+vs_peds <- vs_peds %>%
+  mutate(
+    VSSEQ = row_number(),
+    VSPOS = NA_character_,
+    VSORRESU = case_when(
+      VSTESTCD == "HEIGHT" ~ "cm",
+      VSTESTCD == "WEIGHT" ~ "kg",
+      VSTESTCD == "BMI" ~ "kg/m2",
+      VSTESTCD == "HDCIRC" ~ "cm",
+      TRUE ~ NA_character_
+    ),
+    VSTEST = case_when(
+      VSTESTCD == "HEIGHT" ~ "Height",
+      VSTESTCD == "WEIGHT" ~ "Weight",
+      VSTESTCD == "BMI" ~ "BMI",
+      VSTESTCD == "HDCIRC" ~ "Head Circumference",
+      TRUE ~ NA_character_
+    ),
+    VSSTRESC = as.character(VSSTRESN),
+    VSORRES = VSSTRESC,
+    VSSTRESU = VSORRESU,
+    VSSTAT = NA_character_,
+    VSLOC = NA_character_,
+    VSBLFL = if_else(VISITNUM == 2, "Y", NA_character_),
+    VISITDY = NA_integer_,
+    VSDY = NA_integer_,
+    VSEVAL = NA_character_,
+    EPOCH = "Epoch"
```

Comment:
Is it intentional to leave those columns blank (including `VSBLFL`), should we just remove this part and carry values over from the original VS data?

Reply:
By the way, should `VSBLFL` be replaced by `VSLBOXFL`? In the SDTMIG it is stated that `VSBLFL` is kept only for backward compatibility. See also "4.5.9 Baseline Values" in the SDTMIG.